### PR TITLE
Go tracing documentation

### PR DESCRIPTION
**Pull request by SpennyNDaJets**
## Summary
We support adding a manual trace for Go as well as auto instrumentation for GORM. We are reformatting how the documents for tracing are organized

```
- Backend: Tracing
  - Overview
  - Language 1
    - Overview
    - Manual Tracing
    - Library 1 Auto-Instrumentation
    - Library 2 Auto-Instrumentation
  - Language 2
    - Overview
    - Manual Tracing
    - Library 1 Auto-Instrumentation
    - Library 2 Auto-Instrumentation
  - OLTP
```

With the update of the Go documentation, the file structure looks like:
```
- Backend: Tracing
  - Overview
  - Go
    - Overview
    - Manual Tracing
    - GORM 
  - OLTP
```

## How did you test this change?
Click test docs
- [ ] No extra/missing documentation
- [ ] Correct structure/order
- [ ] Clear and correct instructions

## Are there any deployment considerations?
N/A

## Does this work require review from our design team?
N/A

### CODE REVIEW

**Review by deltaepsilon — APPROVED** [review]

**Comment by SpennyNDaJets** [comment]
Tagging @Vadman97 for ordering with #7247

File: highlight.io/components/QuickstartContent/traces/go/gorm.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+				'Import the Highlight GORM library and call the `SetupGORMTracing` hook with any attributes wanted for context.',
+			code: [
+				{
+					text: `import (
+	"github.com/highlight/highlight/sdk/highlight-go"
+  htrace "github.com/highlight/highlight/sdk/highlight-go/trace"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+if err := htrace.SetupGORMTracing(db, attribute.String(highlight.ProjectIDAttribute, <YOUR_PROJECT_ID>)); err != nil {
```
Comment (Vadman97) [inline highlight.io/components/QuickstartContent/traces/go/gorm.tsx:26]: i think the order of this is confusing.
`htrace.SetupGORMTracing(db, attribute.String(highlight.ProjectIDAttribute, <YOUR_PROJECT_ID>))`
takes a `db` object, but it's not obvious where that comes from. isn't that the `db` you get from `db, err = gorm.Open([DB_SETTINGS])`?

if so, could we refactor this to be

step 1
```go
DB, err = gorm.Open([DB_SETTINGS])
if err := htrace.SetupGORMTracing(db, attribute.String(highlight.ProjectIDAttribute, <YOUR_PROJECT_ID>)); err != nil {
  // ...
}
```
step 2... Call GORM with the trace context
```go
DB.WithContext(ctx).Find(&user)
```

File: highlight.io/components/QuickstartContent/traces/go/gorm.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  // log or handle error
```
Comment (Vadman97) [inline highlight.io/components/QuickstartContent/traces/go/gorm.tsx:27]: good opportunity to use `highlight.RecordError()` here

**Review by Vadman97 — APPROVED** [review]
:ship: feel free to merge this first @SpennyNDaJets and i'll update my PR around your changes

File: highlight.io/components/QuickstartContent/traces/go/gorm.tsx (hunk 1 above)
Comment (SpennyNDaJets) [reply highlight.io/components/QuickstartContent/traces/go/gorm.tsx:26]: Ohh great catch! I'll update that

**Pull request merged**